delete.ps1

```diff
@@ -145,139 +72,211 @@ function Resolve-ZenyaErrorMessage {
+
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "Successfully deleted account [$($account.Username)]. AccountReference: $($outputContext.AccountReference | ConvertTo-Json -Depth 10)."
                     IsError = $false
                 })
         }
         else {
-            Write-Warning "DryRun: Would delete Zenya account $($currentAccount.userName) ($($currentAccount.id))"
+            Write-Warning "DryRun: Would delete account [$($account.Username)]. AccountReference: $($actionContext.References.Account | ConvertTo-Json -Depth 10)."
         }
+        break
     }
     catch {
         $ex = $PSItem
-        if ( $($ex.Exception.GetType().FullName -eq 'Microsoft.PowerShell.Commands.HttpResponseException') -or $($ex.Exception.GetType().FullName -eq 'System.Net.WebException')) {
-            $errorObject = Resolve-HTTPError -Error $ex
-
-            $verboseErrorMessage = $errorObject.ErrorMessage
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
 
-            $auditErrorMessage = Resolve-ZenyaErrorMessage -ErrorObject $errorObject.ErrorMessage
-        }
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+        Write-Verbose "URI: $($splatWebRequest.Uri)"
+        Write-Verbose "Body: $($body)"
 
-        # If error message empty, fall back on $ex.Exception.Message
-        if ([String]::IsNullOrEmpty($verboseErrorMessage)) {
-            $verboseErrorMessage = $ex.Exception.Message
-        }
-        if ([String]::IsNullOrEmpty($auditErrorMessage)) {
-            $auditErrorMessage = $ex.Exception.Message
-        }
-
-        Write-Verbose "Error at Line '$($ex.InvocationInfo.ScriptLineNumber)': $($ex.InvocationInfo.Line). Error: $($verboseErrorMessage)"
-        
-        $success = $false  
-        $auditLogs.Add([PSCustomObject]@{
-                Action  = "DeleteAccount"
-                Message = "Error deleting Zenya account $($currentAccount.userName) ($($currentAccount.id)). Error Message: $auditErrorMessage"
-                IsError = $True
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Error deleting account [$($account.Username)]. AccountReference: $($actionContext.References.Account | ConvertTo-Json -Depth 10). Error Message: $($errorMessage.AuditErrorMessage)"
+                IsError = $true
             })
-    }
-}
 
-# Send results
-$result = [PSCustomObject]@{
-    Success    = $success
-    Account    = $account
-    Auditlogs  = $auditLogs
+        # Throw terminal error
+        throw
+    }
 
-    # Optionally return data for use in other systems
-    ExportData = [PSCustomObject]@{
-        id       = $aRef.id
-        username = $aRef.username
+}
+catch {
```

Comment:
No, because the auditlog already shows a message such as "Error deleting account [$($account.Username)]. Error Message: etc.".
However, this catch is required to run in preview.